lib/checker/checkercomponent.cpp

```diff
@@ -177,6 +182,11 @@ void CheckerComponent::CheckThreadProc()
 
 			checkable->UpdateNextCheck();
 
+			if (updateNextCheck) {
```

Comment:
That variable name feels a bit off here given that there is a function with the same name but calling it is unrelated to that variable.

Would there be harm in always doing `Checkable::OnNextCheckUpdated`? Does other stuff actually depend on the values in Icinga 2 and IDO/Icinga DB going out of sync?

Reply:
The less we do it, the more efficient we are.

Reply:
That wouldn't make a difference for the overdue computation as checkable that have active checks disabled are already excluded by other means (that shouldn't cover the case that they are disabled application-wide, so there it would probably make a small difference): https://github.com/Icinga/icinga2/blob/6fc9038e4d15e48954a2246cf832444d96155c74/lib/icingadb/icingadb-objects.cpp#L2185-L2204

But please do rename that variable into something like `notifyNextCheck`. In the current state, you might suspect a bug because due to the identical name, it would be logical to argue that `checkable->UpdateNextCheck();` should happen inside `if (updateNextCheck)`, which is not the case.